clang/lib/Driver/OffloadBundler.cpp

```diff
@@ -623,6 +623,18 @@ class ObjectFileHandler final : public FileHandler {
     raw_svector_ostream SymbolsOS(SymbolsBuf);
     LLVMContext Context;
 
+    // Set the default value for opaque pointers prior to parsing the bitcode.
+#if !SPIRV_ENABLE_OPAQUE_POINTERS
+    const bool UseOpaquePointers =
+        llvm::any_of(BundlerConfig.TargetNames, [&](const auto &Target) {
+          const Triple TT = getTargetTriple(Target, BundlerConfig);
+          return !TT.isSPIR() && !TT.isSPIRV();
+        });
+    Context.setOpaquePointers(UseOpaquePointers);
```

Comment:
After poking around in more detail (I don't regularly develop this tool), I've come to the conclusion that this code is an elaborate way to set `Context.setOpaquePointers(true)`. The host is included in `TargetNames`, and can't be SPIRV anyways, so there's always a non-SPIR-V module to cause it to be set to true. This of course obscures the failure modes of what would happen if `UseOpaquePointers` were false, since there's no realistic mode where it *could* be false.

Given that SPIRV_ENABLE_OPAQUE_POINTERS is enabled now and we're starting to remove support for typed pointers, I'm not sure there's any need for this PR anymore.

Reply:
Do you mean `LLVMContext Context;` uses opaque pointers by default and don't need to switch to non-opaque mode anymore?
@GeorgeWeb, are you okay to drop this change?

Reply:
https://github.com/intel/llvm/pull/10888 means INTEL_SYCL_OPAQUEPOINTER_READY is now true and therefore everything is always in opaque pointer mode.

Reply:
Always? 
Why do we add ifdef like this:
```c++
#ifndef INTEL_SYCL_OPAQUEPOINTER_READY
  if (!CLANG_ENABLE_OPAQUE_POINTERS_INTERNAL)
    CmdArgs.push_back("-no-opaque-pointers");
  else if ((Triple.isSPIRV() || Triple.isSPIR()) &&
           !SPIRV_ENABLE_OPAQUE_POINTERS)
    CmdArgs.push_back("-no-opaque-pointers");
#endif
```
?
Shouldn't we just remove this code?

Reply:
In principle, it's a cmake option you can turn off. However, I suspect it doesn't actually work anymore. And I also suspect @jsji is likely to start ripping the code out in the near future, after the code bakes for a little bit more (the PR landed only yesterday).

Reply:
Yes, the plan is to remove the code in #ifndef INTEL_SYCL_OPAQUEPOINTER_READY once our downstream code works well.
ETA 1-2 weeks.

Keep it by guarding it within macros just in case we need to compare to INTEL_SYCL_OPAQUEPOINTER_READY=0 in some case for now.

Reply:
> therefore everything is always in opaque pointer mode.

Yes. Also I believe `setOpaquePointers`  is now an assert, so don't call it anymore.

void LLVMContext::setOpaquePointers(bool Enable) const {
  assert(Enable && "Cannot disable opaque pointers");
}

Reply:
I see! So, assuming `INTEL_SYCL_OPAQUEPOINTER_READY` goes away soon-ish then it's all gonna look cleaner. My patch was I guess really a temporary mitigation for an awkward issue and from the insights of this discussion - thanks @jsji and @jcranmer-intel I am happy to close this PR as it seems unneeded for `intel:sycl`. Thanks!

Reply:
@bader If you are convinced too, I am happy to close this given the future direction.